sympy/vector/integrals.py

```diff
@@ -0,0 +1,79 @@
+from sympy.core.basic import Basic
+from sympy.vector import CoordSys3D, Vector
+from sympy.vector.operators import _get_coord_sys_from_expr
+from sympy.integrals import Integral, integrate
+from sympy.core.function import diff
+
+class ParametricIntegral(Basic):
+    """
+    Represents integral of a scalar or vector field
+    over a Parametric Region
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSys3D, ParametricRegion
+    >>> from sympy.abc import t
+
+    >>> C = CoordSys3D('C)
+    >>> curve = ParametricRegion(t, (3*t - 2, t + 1), {t: (1, 2)})
+
+    >>> ParametricIntegral(C.x, curve)
+    0
+    >>> semisphere = ParametricRegion((phi, theta), (2*sin(phi)*cos(theta), 2*sin(phi)*sin(theta), 2*cos(phi))
+                            {theta: (0, 2*pi), phi: (0, pi/2)})
+    >>> ParametricIntegral(C.z, semisphere)
+    8*pi
+
+    >>> ParametricIntegral(C.j - C.k, ParametricRegion((r, theta), (r*cos(theta), r*sin(theta))))
+    ParametricIntegral(C.j - C.k, ParametricRegion((r, theta), (r*cos(theta), r*sin(theta))))
+
+    """
+    
+    def __new__(cls, field, parametricregion):
+        obj = super().__new__(cls, field, parametricregion)
+        return obj
+
+    def eval(self):
+        parametricregion = self.parametricregion
+        field = self.field
+        
+        coord_sys = _get_coord_sys_from_expr(field)
+        
+        if len(coord_sys) > 1:
+            raise ValueError
+        
+        coord_sys = next(iter(coord_sys))
```

Comment:
Since no coordinate system is associated with the parametric region, we use the base scalars of the field. But if the field does not have any basescalar or vector, we cannot derive coordinate system. This breaks the logic

Example:
```python
>>> p = ParametricRegion(t, (3*t - 2, t + 1), {t: (1, 2)}).
>>> I = ParametricIntegral(C.x + C.y, p)
>>> I.eval()
5*sqrt(10)
>>>
>>> I = ParametricIntegral(1, p) ##We cannot extract voordinate system from 1
StopIteration
```

Reply:
My opinion is that we should equip base scalars and base vector (fields) to the `ParametricRegion` if we want to use its intrinsic scalars and vector field components.
I don't think that it makes much sense to have vector integral like `ParametricIntegral(C.x*C.i, p)` where C.x is linked to the base scalars of the parametric region and `C.i` is linked to embedding cartesian coordinate system.

Reply:
Agreed. But if we allow the coordsys parameter to be optional and None by default in ParametricRegion class, same situation can arise.

Reply:
@Upabjojr, your thoughts on this?

Reply:
Probably it's better to separate vector and parametric integrals. At least for now.

Reply:
I am not sure what you mean here. Do you mean a separate class for handling vector fields? Please explain.
I think it will be better if we associate some default coordinate system with base scalars with a ParamRegion Object?

Reply:
What about temporarily ignoring coordinate systems? We can go back to them in the future.

Reply:
`ParametricIntegral(C.i, ... )` is a vector, while `ParametricIntegral(C.x, ... )` is a scalar. I would expect different classes for vectors and scalars. For example, vectors should be endowed with methods for vector operations.

Reply:
> I would expect different classes for vectors and scalars. 

I think it will be easier for users if we have a common class for vector and scalar fields. Also, the calculation is very similar in both cases. It just differs whether we take dot product with field or multiply it with a modulus of the normal vector to the surface

If we decide to make separate classes,  we also should have a class that returns an object of ParametricScalarIntegral or ParametricVectorIntegral depending on the nature of the field.

> vectors should be endowed with methods for vector operations

Does it mean applying curl and divergence operators as used in Green's theorem and the Divergence theorem? It would be nice to have them.  but we can easily apply them to the field without any method too.
```python
ParametricIntegral(curl(C.x*C.y*C.i), surface))
```

Reply:
What about using class inheritance?

Reply:
Yes, we can make a 'ParametricRegion` class and make two subclasses: `ScalarParametricIntegral` and `VectorParametricIntegral`.
But I think It would be easier if a user can use a single class for both scalar and vector fields.

Reply:
> ParametricIntegral(C.i, ... ) is a vector, while ParametricIntegral(C.x, ... ) is a scalar
How can `ParametricIntegral(C.i, ...)` be a vector? I think it is assumed in most textbooks that the dot product of vector field and position vector of the Parametric region takes place and the result is integrated along the region.

Reply:
Maybe we can have a function `vectorintegrate` which returns the object of ParametricVectorIntegral or ParametricScalarIntegral depending on the nature of the field. 

This can work for other types of integrals too. Suppose we implement a class ImplicitIntegral to calculate the integral over an implicit region. The `vectorintegrate` function can then return an object of ImplicitIntegral object by identifying the region is defined implicitly.

@Upabjojr

Reply:
There are various ways to dispatch. Yes, in general the users should not have to choose between the subclass... there should be one function and then the code should decide which class to use.

The point of having subclasses is that the result of the vector integration may be a vector or a scalar, depending on what and how you are integrating.

The vector object should behave like a vector (e.g. have methods of vectors).

Reply:
> The point of having subclasses is that the result of the vector integration may be a vector 

I do not see how this is possible.

Reply:
I think this is possible if the user want to perform cross product of vector field with normal vector instead of the usual dot product.

Reply:
Well, integrating velocity over time is a simple example. Though it's probably already handled by the current `Integral` class. But you see, this case could be a case where you may want to subclass `Integral` in order to supply it with the vector methods.

Reply:
But indeed, probably we don't need this case, it's pretty rare that some users may wish to use it.

Reply:
Although it is a kind of vector integral, it is not integrating a vector over a region. As you said, the `integrate` function is able to handle such kind of integrals (integral over variables). But if we wish to handle such case, we can allow passing a tuple of variables in place of the parametric region.